octobot_trading/personal_data/portfolios/portfolio_manager.py

```diff
@@ -119,14 +120,19 @@ async def _reset_portfolio(self):
     def _refresh_simulated_trader_portfolio_from_order(self, order):
         """
         Handle a balance update from an order request when simulating
+        Catch a PortfolioNegativeValueError when calling portfolio update method and returns False if raised
         :param order: the order that should update portfolio
         :return: True if the portfolio was updated
         """
-        if order.is_filled():
-            self.portfolio.update_portfolio_from_filled_order(order)
-        else:
-            self.portfolio.update_portfolio_available(order, is_new_order=False)
-        return True
+        try:
+            if order.is_filled():
+                self.portfolio.update_portfolio_from_filled_order(order)
+            else:
+                self.portfolio.update_portfolio_available(order, is_new_order=False)
+            return True
+        except errors.PortfolioNegativeValueError as portfolio_negative_value_error:
+            self.logger.error(f"Failed to update portfolio : {portfolio_negative_value_error}")
```

Comment:
I think this should be a self.logger.exception so that we get a callstack when it happens and it will help to figure out why

Reply:
Right